atomos.framework/src/main/java/org/atomos/framework/AtomosRuntime.java

```diff
@@ -293,18 +296,19 @@ static Framework launch(Map<String, String> frameworkConfig) throws BundleExcept
 	 * @return a map of the configuration specified by the args
 	 */
 	static Map<String, String> getConfiguration(String[] args) {
-		Map<String, String> config = new HashMap<>();
-		if (args != null) {
-			for (String arg : args) {
-				int equals = arg.indexOf('=');
-				if (equals != -1) {
-					String key = arg.substring(0, equals);
-					String value = arg.substring(equals + 1);
-					config.put(key, value);
-				}
-			}
-		}
-		return config;
+		return Optional.ofNullable(args)
```

Comment:
I have a few comments/questions about this change.

1. What prompted the change?  It does not appear to fix a bug with the original code and I argue that it is not quite as understandable as the original.
1. I've been stuck using pre Java-8 so long that I'm still relatively new to lamdas and functional programming in Java so I could be biased - I'm not sure this use of lambdas and streams is warranted over the original for-each code.  First it creates an extra List<String> in order to get a stream.  Then it does some bit of double work to filter out based on presence of `'='` char, but then in the next stage of the pipeline it has to do the same work to locate the `'='` character again

Reply:
I agree it is not  better readable! I just played a bit with atomos and tried to understand everything.